pkg/kubectl/cmd/attach.go

```diff
@@ -196,17 +206,37 @@ func (p *AttachOptions) Run() error {
 	}
 	if p.Stdin {
 		t.In = p.In
-		if tty && !t.IsTerminal() {
+		if tty && !t.IsTerminalIn() {
 			tty = false
 			fmt.Fprintln(p.Err, "Unable to use a TTY - input is not a terminal or the right kind of file")
 		}
+	} else {
+		p.In = nil
 	}
 	t.Raw = tty
 
-	fn := func() error {
-		if tty {
-			fmt.Fprintln(p.Out, "\nHit enter for command prompt")
+	var sizeQueue term.TerminalSizeQueue
+	if tty {
+		size := t.GetSize()
+		if size != nil {
+			// fake resizing +1 and then back to normal so that attach-detach-reattach will result in the
+			// screen being redrawn
+			sizePlusOne := *size
+			sizePlusOne.Width++
+			sizePlusOne.Height++
+
+			// this call spawns a goroutine to monitor/update the terminal size
+			sizeQueue = t.MonitorSize(&sizePlusOne, size)
 		}
+
+		// unset p.Err if it was previously set because both stdout and stderr go over p.Out when tty is
+		// true
+		p.Err = nil
+	}
+
+	fn := func() error {
+		fmt.Fprintln(p.Out, "If you don't see a command prompt, try pressing enter.")
```

Comment:
I would expect this on stderr. If you parse the command output in a script, you get this line otherwise.